### PR TITLE
[DOPS-917] Introduction of makefile build-docker target

### PR DESCRIPTION
**Pull request by phoinixgrr**
#### Summary

Addition of Makefile target that takes care of building a docker image. 
Example : 
```
make docker-build tag=rtcd:latest
docker build -f build/Dockerfile . -t rtcd:latest
[+] Building 21.3s (15/15) FINISHED
 => [internal] load build definition from Dockerfile                                                                                               0.0s
 => => transferring dockerfile: 37B                                                                                                                0.0s
 => [internal] load .dockerignore                                                                                                                  0.0s
 => => transferring context: 2B                                                                                                                    0.0s
 => [internal] load metadata for gcr.io/distroless/static@sha256:d6fa9db9548b5772860fecddb11d84f9ebd7e0321c0cb3c02870402680cc315f                  0.6s
 => [internal] load metadata for docker.io/library/golang:1.17.9-alpine@sha256:5c2fcfeb58ad9d4948d94e7b2d0432a9bc38bee0f8dfb41d383f38a18b75c38d    0.0s
 => [builder 1/6] FROM docker.io/library/golang:1.17.9-alpine@sha256:5c2fcfeb58ad9d4948d94e7b2d0432a9bc38bee0f8dfb41d383f38a18b75c38d              0.0s
 => [runner 1/3] FROM gcr.io/distroless/static@sha256:d6fa9db9548b5772860fecddb11d84f9ebd7e0321c0cb3c02870402680cc315f                             0.0s
 => [internal] load build context                                                                                                                  0.0s
 => => transferring context: 37.31kB                                                                                                               0.0s
 => CACHED [builder 2/6] RUN apk update && apk add make                                                                                            0.0s
 => CACHED [builder 3/6] RUN mkdir -p /rtcd                                                                                                        0.0s
 => [builder 4/6] ADD . /rtcd                                                                                                                      0.0s
 => [builder 5/6] WORKDIR /rtcd                                                                                                                    0.0s
 => [builder 6/6] RUN make build                                                                                                                  20.5s
 => CACHED [runner 2/3] COPY --from=builder /rtcd/dist/rtcd /rtcd/rtcd                                                                             0.0s
 => CACHED [runner 3/3] WORKDIR /rtcd                                                                                                              0.0s
 => exporting to image                                                                                                                             0.0s
 => => exporting layers                                                                                                                            0.0s
 => => writing image sha256:19476885ac5b1e105064428c6bc8004c596e2b32b2bb8077fd850a6bc3f5d4a3                                                       0.0s
 => => naming to docker.io/library/rtcd:latest                                                                                                     0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
```

#### Ticket Link

https://mattermost.atlassian.net/browse/DOPS-917

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,16 +23,30 @@ golangci-lint: ## Run golangci-lint on codebase
...
 	env GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o dist/rtcd -ldflags '$(LDFLAGS)' -mod=readonly -trimpath ./cmd/rtcd
 
+.PHONY: docker-build
+docker-build:
+	@if ! [ -x "$$(command -v docker)" ]; then \
+		echo "Docker is not installed. Please see https://docs.docker.com/get-docker for installation instructions."; \
+		exit 127; \
+	fi; \
+
+	@[ "${tag}" ] || ( echo "build-docker requires a tag. Set a tag with: \"make docker-build tag=rtcd:my_tag\""; exit 1 )
```
Comment (pfltdv) [inline Makefile:42]: Small Fix.
```suggestion
	@[ "${tag}" ] || ( echo "docker-build requires a tag. Set a tag with: \"make docker-build tag=rtcd:my_tag\""; exit 1 )
```

File: Makefile [hunk 2]
```diff
@@ -21,16 +23,30 @@ golangci-lint: ## Run golangci-lint on codebase
...
+
+	docker build -f build/Dockerfile . -t $(tag)
```
Comment (spirosoik) [inline Makefile:44]: nice, not for now but in a later commit let's make [https://github.com/mattermost/rtcd/blob/master/build/Dockerfile#L4] an `ARG` so we can change this easily with `--build-arg`

**Review by spirosoik — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (pfltdv) [inline Makefile:42]: Is it possible to exit with another code. It is easy to confuse exit code `1` with general errors. Suggestion `128`- `Invalid Argument`

Comment (phoinixgrr) [reply Makefile:42]: Actually `127` is `command not found`, which is exactly the case here: 
http://www.bic.mni.mcgill.ca/~dale/helppages/BashGuide/advshell/exitcodes.html

Comment (phoinixgrr) [reply Makefile:42]: disregard my comment, I see you are commenting  online 42

Comment (phoinixgrr) [reply Makefile]: Rebased and pushed changes requested @pfltdv

Comment (phoinixgrr) [reply Makefile]: Rebased and push Fix 
Thanks @pfltdv

**Review by pfltdv — APPROVED** [review]
🏅 Nice work @phoinixgrr

**Review by spirosoik — DISMISSED** [review]

**Comment by phoinixgrr** [comment]
@spirosoik  @pfltdv  Had to drop your reviews, since I recommited signed commits to honor new repository policy requiring signing commits.

**Review by spirosoik — APPROVED** [review]

**Review by pfltdv — APPROVED** [review]

**Pull request merged**